app.js

```diff
@@ -506,256 +506,148 @@ VIEWER.consumeForGeoJSON = async function(dataURL) {
-                                return canvas.navPlace
-                            }
-                        })
-                }
-                //Combine them together so that they are all drawn on the web map
-                geoJSONFeatures = [...geos, ...structuresGeos, ...itemsGeos]
+                let geos = getResourceProperties("manifest")
+                let [structureGeo, itemsGeos] = await getStructures(VIEWER.resource)
+                geoJSONFeatures = [...geos, ...structureGeo, ...itemsGeos]
                 return geoJSONFeatures
             break
             case "Canvas":
-                let canvasGeo = {}
-                if (VIEWER.resource.hasOwnProperty("navPlace")) {
-                    if (VIEWER.resource.navPlace.features) {
-                        VIEWER.resource.navPlace.features = VIEWER.resource.navPlace.features.map(f => {
-                            //FIXME support referenced Features even though the spec encourages embedded Features?
-                            if (!f.properties.thumbnail) {
-                                //Then lets grab the image URL from the annotation of the first Canvas item if available.  
-                                if(VIEWER.resource.thumbnail){
-                                    f.properties.thumbnail = VIEWER.resource.thumbnail
-                                }
-                                else if (VIEWER.resource.hasOwnProperty("items") && VIEWER.resource.items.length && VIEWER.resource.items[0].items.length && VIEWER.resource.items[0].items[0].items.length) {
-                                    if (VIEWER.resource.items[0].items[0].items[0].body) {
-                                        let thumburl = VIEWER.resource.items[0].items[0].items[0].body.id ?? ""
-                                        f.properties.thumbnail = [{"id":thumburl}]
-                                    }
-                                }
-                            }
-                            if (!f.properties.hasOwnProperty("summary")) {
-                                f.properties.summary = VIEWER.resource.summary ?? ""
-                            }
-                            if (!f.properties.hasOwnProperty("label")) {
-                                f.properties.label = VIEWER.resource.label ?? ""
-                            }
-                            if (!f.properties.hasOwnProperty("canvas")) {
-                                f.properties.canvas = VIEWER.resource["@id"] ?? VIEWER.resource["id"] ?? "Yikes"
-                            }
-                            return f
-                        })
-                    }
-                    geoJSONFeatures = VIEWER.resource.navPlace
-                    return geoJSONFeatures
-                }
+                let canvas_geos = getResourceProperties("canvas")
+                //Canvas is special because it does not need the return value to be an Array, so the Array must be unpacked
```

Comment:
Hmm I question what this is, but everything works.  A better note might be needed.